packages/flutter/test/material/reorderable_list_test.dart

```diff
@@ -112,6 +113,19 @@ void main() {
         expect(listItems, orderedEquals(<String>['Item 2', 'Item 3', 'Item 4', 'Item 1']));
       });
 
+      testWidgets('properly reorders with a footer', (WidgetTester tester) async {
```

Comment:
It seems like this test is failing presubmit checks.

Reply:
@ueman Could you explain what you changed here to get it to pass? I wasn't able to see the differences because you force-pushed, so the commit history was lost

Reply:
Sure. I've used the test ("properly reorders with a header") directly above it as a base for my test. 
What I've changed is that I've wrapped the footer in a SizedBox to give it the same width as normal list entry, so that I know how wide it is because the drop area is after the footer (which is the behaviour in a reversed horizontal list). Then I've also changed the footer text from "Footer Text" to just "Footer" to make sure it fits inside the SizedBox.

Reply:
```dart
testWidgets('properly reorders with a footer', (WidgetTester tester) async {
        await tester.pumpWidget(build(header: const Text('Footer Text'), scrollDirection: Axis.horizontal));
        expect(find.text('Footer Text'), findsOneWidget);
        expect(listItems, orderedEquals(originalListItems));
        await longPressDrag(
          tester,
          tester.getCenter(find.text('Item 1')),
          tester.getCenter(find.text('Item 4')) + const Offset(itemHeight * 2, 0.0),
        );
        await tester.pumpAndSettle();
        expect(find.text('Footer Text'), findsOneWidget);
        expect(listItems, orderedEquals(<String>['Item 2', 'Item 3', 'Item 4', 'Item 1']));
        await tester.pumpWidget(build(header: const Text('Footer Text'), scrollDirection: Axis.horizontal));
        await longPressDrag(
          tester,
          tester.getCenter(find.text('Item 4')),
          tester.getCenter(find.text('Item 3')),
        );
        expect(find.text('Footer Text'), findsOneWidget);
        expect(listItems, orderedEquals(<String>['Item 2', 'Item 4', 'Item 3', 'Item 1']));
      });
```

Reply:
I do think that there is a real issue with your implementation. The intermediary check was accidentally removed:
```
        await longPressDrag(
          tester,
          tester.getCenter(find.text('Item 1')),
          tester.getCenter(find.text('Item 4')) + const Offset(itemHeight * 2, 0.0),
        );
        await tester.pumpAndSettle();
        expect(find.text('Footer Text'), findsOneWidget);
        expect(listItems, orderedEquals(<String>['Item 2', 'Item 3', 'Item 4', 'Item 1']));
```

In a real app, the behavior is prevalent as well. You cannot drag an item from the center to the horizontal end of the list.

![reorderable_bug](https://user-images.githubusercontent.com/27032613/75373665-8ee9cf80-587f-11ea-9612-2b5becd98116.gif)

Could you please fix this issue?

Reply:
Yes. That could take however a few days as I'm busy.
On a side note, should the additional space after list still be there if a footer is existent? I didn't find anything about that on [https://material.io/components/lists/#types](https://material.io/components/lists/#types).

Reply:
I'm not sure what space you're referring to. Would this be space between the last item and the edges of the widget?

Reply:
I'm referring to the finalDropArea here:  https://github.com/flutter/flutter/blob/0b8abb4724/packages/flutter/lib/src/material/reorderable_list.dart#L544
It looks kinda weird if it's before the footer, so I left it after the footer. This also causes the effect described by you that you can't drag an item between the last item of the list and the footer. To order the item to the last position you have to drag it to the finalDropArea.
In the following screenshot I've marked finalDropArea in red.
![Simulator Screen Shot - iPhone 11 Pro Max - 2020-02-26 at 20 16 00](https://user-images.githubusercontent.com/1270149/75378920-d9496b80-58d4-11ea-9dac-4353462eea56.png)